datafusion/expr/src/logical_plan/ddl.rs

```diff
@@ -341,29 +341,8 @@ pub struct CreateFunctionBody {
     pub language: Option<Ident>,
     /// IMMUTABLE | STABLE | VOLATILE
     pub behavior: Option<Volatility>,
-    /// AS 'definition'
-    pub as_: Option<DefinitionStatement>,
-    /// RETURN expression
-    pub return_: Option<Expr>,
-}
-
-#[derive(Clone, PartialEq, Eq, Hash, Debug)]
-pub enum DefinitionStatement {
-    SingleQuotedDef(String),
-    DoubleDollarDef(String),
-}
-
-impl From<sqlparser::ast::FunctionDefinition> for DefinitionStatement {
-    fn from(value: sqlparser::ast::FunctionDefinition) -> Self {
-        match value {
-            sqlparser::ast::FunctionDefinition::SingleQuotedDef(s) => {
-                Self::SingleQuotedDef(s)
-            }
-            sqlparser::ast::FunctionDefinition::DoubleDollarDef(s) => {
-                Self::DoubleDollarDef(s)
-            }
-        }
-    }
+    /// RETURN or AS function body
+    pub function_body: Option<Expr>,
```

Comment:
👍

Reply:
FYI @milenkovicm -- sqlparser changed how the body was represented so now this changes as well to a generic Expr

Reply:
Thanks for letting me know, @alamb 

I just wonder if this change will diverge sqlparser from Postgres? 

I guess function like this 

```sql
CREATE FUNCTION strlen(name TEXT)
    RETURNS int LANGUAGE plrust AS
$$
    Ok(Some(name.unwrap().len() as i32))
$$;

```

Would not be possible without wrapping body with string

Reply:
It seems like we may need to add `$$` string parsing support in sqlparser. I'll try and file a ticket later todday

Reply:
IIRC we should previously support `$$` function body. If it fails to parse now, it's a regression.

Reply:
It used to work, https://github.com/sqlparser-rs/sqlparser-rs/pull/1185

Reply:
I added a test and made this syntax work in this pr in 84468515808991159632aac8db0f68e7e039419b